src/providers/prize-values.tsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { createContext, useEffect, useContext, useState } from 'react';
+import { usePoolData } from './pools';
+import { usePoolChainData } from './pools-chain';
+import { usePoolsERC20Balance } from './pools-erc-balance';
+import { useExchange } from './exchange';
+import { useNetworkProvider } from './ethers';
+import { calculateEstimatedPoolPrize } from '../utils/calculateEstimatedPrizePool';
+import { ethers, BigNumber } from 'ethers';
+import { derivedEthToEthAlreadyNormalized, EthToUsd, derivedEthToEth } from '../utils/conversions';
+import { addEmitHelper } from 'typescript';
+
+const PrizeValContext = createContext<string[]>([]);
+
+export const PrizeProvider: React.FC = ({ children }) => {
+  const {
+    ethToUsd,
+    tokenExchange: { initialized, exchangeDict },
+  } = useExchange();
+  const pools = usePoolData();
+  const poolsChainData = usePoolChainData();
+  const poolsErc20Balance = usePoolsERC20Balance();
+  const provider = useNetworkProvider();
+  const [prizes, setPrizes] = useState<string[]>([]);
+  useEffect(() => {
+    (async () => {
+      if (!provider) return;
+      const chainId = (await provider.getNetwork()).chainId;
+      if (poolsErc20Balance.length !== pools.length || (!initialized && chainId == 1)) return;
+      const prizeVals = [];
+      console.log({ ethToUsd: ethers.utils.formatUnits(ethToUsd, 8) });
+      for (let i = 0; i < pools.length; i++) {
+        const prizeForPool = calculateEstimatedPoolPrize({
+          tokenDecimals: parseInt(pools[i].underlyingCollateralDecimals, 10),
+          awardBalance: poolsChainData[i].awardBalance,
+          supplyRatePerBlock: poolsChainData[i].supplyRatePerBlock,
+          prizePeriodRemainingSeconds: poolsChainData[i].secondsRemaining,
+          poolTotalSupply: BigNumber.from(pools[i].ticketSupply).add(BigNumber.from(pools[i].sponsorshipSupply)),
+        });
+        console.log(pools[i].underlyingCollateralSymbol);
+        console.log('------------------');
+        if (chainId == 4) {
+          console.log(ethers.utils.formatEther(prizeForPool));
+          prizeVals.push(truncate(ethers.utils.formatEther(prizeForPool)));
+
+          // ^ just following PT front-end which seems to do this for rinkeby
+          continue;
+        }
+        let poolPrizeEth = derivedEthToEthAlreadyNormalized(
+          prizeForPool,
+          exchangeDict[pools[i].underlyingCollateralContract.address.toLowerCase()].derivedEth,
+        );
+        for (const balance of poolsErc20Balance[i]) {
```

Comment:
this is basically adding the value of the lootbox prizes and external erc20 prizes (which are separate I think... lootbox is an erc721 external prize which you can attack erc20 tokens to).